wallet/src/de/schildbach/wallet/ui/dashpay/NotificationsForUserLiveData.kt

```diff
@@ -13,13 +12,17 @@ class NotificationsForUserLiveData(walletApplication: WalletApplication, platfor
         this.query = userId
         GlobalScope.launch {
             val results = arrayListOf<NotificationItem>()
-            val contactRequests = platformRepo.searchContacts("", UsernameSortOrderBy.DATE_ADDED)
+            val contactRequests = platformRepo.searchContacts("", UsernameSortOrderBy.DATE_ADDED, true)
 
-            if(contactRequests.data != null) {
-                contactRequests.data.filter {
-                    cr -> cr.dashPayProfile.userId == userId
+            if (contactRequests.data != null) {
+                contactRequests.data.filter { cr ->
+                    cr.dashPayProfile.userId == userId
                 }.forEach {
-                    results.add(NotificationItem(it))
+                    results.add(NotificationItemContact(it))
+                    if (it.type == UsernameSearchResult.Type.CONTACT_ESTABLISHED) {
+                        val invitationItem = if (it.incoming) it.copy(toContactRequest = null) else it.copy(fromContactRequest = null)
+                        results.add(NotificationItemContact(invitationItem, isInvitationOfEstablished = true))
+                    }
```

Comment:
Is it the right approach?
I generate "invitationItem" for each established contacts so that it could be displayed in the history.

Reply:
very cool.  Splitting into two items seems like a good way to do it.  But I will need to review this more to get a better idea of how it will work.